CMakeLists.txt

```diff
@@ -1,77 +1,97 @@
-project(json-schema-validator
+project(nlohmann_json_schema_validator
     LANGUAGES CXX)
 
-set(PROJECT_VERSION 2.0.0)
+set(PROJECT_VERSION 2.1.0)
 
 cmake_minimum_required(VERSION 3.2)
 
 option(BUILD_TESTS      "Build tests"    ON)
 option(BUILD_EXAMPLES   "Build examples" ON)
 
-# if used as a subdirectory just define a json-hpp-target as add_library(json-hpp INTERFACE)
-# and associate the path to json.hpp via target_include_directories()
-if(NOT TARGET json-hpp)
-    set(NLOHMANN_JSON_DIR "" CACHE STRING "path to json.hpp")
-
-    # find nlohmann's json.hpp
-    find_path(JSON_HPP nlohmann/json.hpp
-        PATHS
-            ${NLOHMANN_JSON_DIR}
-            ${CMAKE_BINARY_DIR}/${NLOHMANN_JSON_DIR}) # in case it is a relative path
-
-    # get the full, real path
-    get_filename_component(NLOHMANN_JSON_REALPATH ${JSON_HPP} REALPATH)
-
-    if(NOT EXISTS ${NLOHMANN_JSON_REALPATH}/nlohmann/json.hpp)
-        message(FATAL_ERROR "please set NLOHMANN_JSON_DIR to a path in which NLohmann's json.hpp can be found. Looking for nlohmann/json.hpp in '${NLOHMANN_JSON_REALPATH}")
-    endif()
-
-    # create an interface-library for simple cmake-linking
-    add_library(json-hpp INTERFACE)
-    target_include_directories(json-hpp
-        INTERFACE
-            ${NLOHMANN_JSON_REALPATH})
-endif()
-
-# and one for the validator
-add_library(json-schema-validator
+# the library
+add_library(nlohmann_json_schema_validator
     src/json-schema-draft7.json.cpp
     src/json-uri.cpp
     src/json-validator.cpp)
 
-set_target_properties(json-schema-validator
+target_include_directories(nlohmann_json_schema_validator
+    PUBLIC
+        $<INSTALL_INTERFACE:include>
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src>)
+
+target_compile_features(nlohmann_json_schema_validator
+    PUBLIC
+        cxx_range_for) # for C++11 - flags
+
+set_target_properties(nlohmann_json_schema_validator
                       PROPERTIES
                           VERSION ${PROJECT_VERSION}
                           SOVERSION 1)
 
-install(TARGETS json-schema-validator
-        LIBRARY DESTINATION lib
-        ARCHIVE DESTINATION lib
-        RUNTIME DESTINATION bin)
+# here we decice how nlohmann::json is found and used to build this project
+
+# first, check whether a nlohmann_json::nlohmann_json target exists already
+# -> we are used as a sub-directory from within another project
+if(TARGET nlohmann_json::nlohmann_json)
+    message(STATUS "Found nlohmann_json::nlohmann_json-target - linking with it")
+    target_link_libraries(
+        nlohmann_json_schema_validator
+        PUBLIC nlohmann_json::nlohmann_json)
+
+elseif(TARGET nlohmann_json) # or nlohmann_json, we are used a sub-project next to nlohmann-json's git repo
```

Comment:
This `elseif(TARGET nlohmann_json)` is not needed. When nlohmann_json is included as a sub-directory it creates the alias `nlohmann_json::nlohmann_json`.  When the nlohmann_json project creates the `nlohmann_json` library it always creates the `nlohmann_json::nlohmann_json` library as well.